### DISCUSSION TITLE
Guidance on cloning async_nats::Client

### DISCUSSION THREAD

**Discussion by jaskij** [nats-io/nats.rs#622] [Q&A] [answered]
As in the topic - I need some guidance on sharing the client between multiple async tasks.

Is it safe to call `connect()` once and then pass clones to actual tasks?

Do the individual tasks have to keep their clones around? The way I see it, I would
1. create and keep the `Client` in `main()`
2. pass cloned instances to individual tasks on their creation
3. a subscriber task could create the `Subscriber` and discard it's `Client` clone when it's not needed

**Comment by stevelr** [comment, marked as answer]
Clients are lightweight objects and you can clone as many as you need, and let rust drop them whenever you don't need them. 

Clients and Subscriptions share the same tcp connection created by the call to connect().
